### PR TITLE
GitHub actions migration followup

### LINKED ISSUE #3384: Missing features in GitHub Actions build
This is a quick note of the things I want to improve in the GitHub Actions build:

- [x] when building commits on the master branch we should also test the database extension (currently we do not) and submit coverage to coveralls
- [ ] therefore it could make sense to have a single workflow file and use conditions to disable steps (such as build and upload of snapshot releases when checking a PR)
- [x] we need to test for more than JRE 8 (add JRE 14)
- [x] the coveralls token needs to be available in jobs triggered by PRs from forks (https://github.com/OpenRefine/OpenRefine/pull/3385/checks?check_run_id=1500751010)

### PR DESCRIPTION
**Pull request by wetneb**
This addresses various issues that have arisen after the migration from Travis to GitHub actions.
Closes #3384.

### CODE REVIEW

**Comment by wetneb** [comment]
I ended up deciding against merging the two workflows, because it would make the workflow file quite unreadable with an entanglement of conditional steps. (But I do not have strong feelings about that). We only want to publish the snapshot release once (so it would not make sense to duplicate this process for Java 8 and Java 14).

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
+[![Join the chat at https://gitter.im/OpenRefine/OpenRefine](https://badges.gitter.im/OpenRefine/OpenRefine.svg)](https://gitter.im/OpenRefine/OpenRefine) ![Java CI](https://github.com/OpenRefine/OpenRefine/workflows/Java%20CI/badge.svg) [![Coverage Status](https://coveralls.io/repos/github/OpenRefine/OpenRefine/badge.svg?branch=master)](https://coveralls.io/github/OpenRefine/OpenRefine?branch=master) [![Translation progress](https://hosted.weblate.org/widgets/openrefine/-/svg-badge.svg)](https://hosted.weblate.org/engage/openrefine/?utm_source=widget) [![Total alerts](https://img.shields.io/lgtm/alerts/g/OpenRefine/OpenRefine.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/OpenRefine/OpenRefine/alerts/)
```
Comment (tfmorris) [inline README.md:3]: I'm not sure "Java CI" is clearer than "Build Status", but I guess I don't feel that strongly about it.

**Review by tfmorris — APPROVED** [review]
Thanks for putting all the work into this!

**Comment by tfmorris** [comment]
Note that all pending PRs will need to be rebased on the `master` branch post-merge of this PR to get their CI jobs running.

**Pull request merged**